main/src/ca/uwaterloo/flix/language/phase/HtmlDocumentor.scala

```diff
@@ -0,0 +1,640 @@
+/*
+ * Copyright 2023 Holger Dal Mogensen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.{Ast, SourceLocation, Symbol, Type, TypedAst}
+import ca.uwaterloo.flix.language.fmt.{FormatType, SimpleType}
+
+import java.io.IOException
+import java.nio.file.{Files, Path, Paths}
+import scala.collection.parallel.CollectionConverters.IterableIsParallelizable
+import com.github.rjeschke.txtmark
+import scala.io.Source
+
+/**
+  * A phase that emits a JSON file for library documentation.
+  */
+object HtmlDocumentor {
+
+  /**
+    * The "Pseudo-name" of the root namespace.
+    */
+  val RootNS: String = "Prelude"
+
+  /**
+    * The directory where to write the ouput.
+    */
+  val OutputDirectory: Path = Paths.get("./build/doc")
+
+  /**
+    * The path to the the stylesheet, relative to the resources folder.
+    */
+  val Stylesheet: String = "/doc/styles.css"
+
+  def run(root: TypedAst.Root)(implicit flix: Flix): TypedAst.Root = flix.phase("HtmlDocumentor") {
+    //
+    // Determine whether to generate documentation.
+    //
+    if (!flix.options.documentorHtml) {
+      return root
+    }
+
+    clearOutputDirectory()
+    writeStyles()
+    val modules = splitModules(root)
+    modules.par.foreach {
```

Comment:
Lets not write to the disk in parallel. Better to just do it one file at a time.

Reply:
How come?

Reply:
It is slower.